### PR TITLE
[Enhancement]avoid Aws::Client::ClientConfiguration default ctor which has a remote access

### PR DESCRIPTION
**Pull request by zombee0**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0

### CODE REVIEW

File: be/src/fs/fs_s3.cpp [hunk 1]
```diff
@@ -182,7 +95,7 @@ std::shared_ptr<Aws::Auth::AWSCredentialsProvider> S3ClientFactory::_get_aws_cre
 
     if (!aws_cloud_credential.iam_role_arn.empty()) {
         // Do assume role
-        Aws::Client::ClientConfiguration clientConfiguration{};
+        Aws::Client::ClientConfiguration clientConfiguration = S3ClientFactory::getClientConfig();
```
Comment (kevincai) [inline be/src/fs/fs_s3.cpp:98]: is this the only key change in this PR?

Comment (zombee0) [reply be/src/fs/fs_s3.cpp:98]: Yes, other changes are in ut, also the same change.

**Comment by dirtysalt** [comment]
Can you describe which remote access?  w

**Review by dirtysalt — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Review by Smith-Cruise — APPROVED** [review]

**Comment by zombee0** [comment]
> Can you describe which remote access? w

let me show u code in default ClientConfiguration ctor:
`ClientConfiguration::ClientConfiguration()
{
    this->disableIMDS = false;
    setLegacyClientConfigurationParameters(*this);
    retryStrategy = InitRetryStrategy();

    if (!this->disableIMDS &&
        region.empty() &&
        Aws::Utils::StringUtils::ToLower(Aws::Environment::GetEnv("AWS_EC2_METADATA_DISABLED").c_str()) != "true")
    {
        auto client = Aws::Internal::GetEC2MetadataClient();
        if (client)
        {
            region = client->GetCurrentRegion();
        }
    }
    if (!region.empty())
    {
        return;
    }
    region = Aws::String(Aws::Region::US_EAST_1);
    setConfigFromEnvOrProfile(*this);
}`

**Review by DorianZheng — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**